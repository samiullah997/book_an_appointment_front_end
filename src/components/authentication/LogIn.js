/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { SigninDetailsApi } from '../../redux/reducer/user';

const LogIn = () => {
  const [showMessage, setShowMessage] = useState('');
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassowrd] = useState('');
  const [loading, setLoading] = useState(false);
  const state = useSelector((state) => state.UserReducer);
  const { loggedIn } = state;
  const loginUser = (e) => {
    e.preventDefault();
    setLoading(true);
    if (email === '' || password === '') return;
    const newUser = {
      email,
      password,
    };
    dispatch(SigninDetailsApi(newUser));
    setEmail('');
    setPassowrd('');
  };

  useEffect(() => {
    if (loggedIn === 'in') {
      setLoading(false);
      navigate('/user/dashboard', { replace: true });
    }
    if (loggedIn === 'err') {
      setLoading(false);
      setShowMessage('Your credential is not correct.');
      navigate('/', { replace: true });
    }
  }, [state]);

  return (
    <div className="flex h-screen w-full items-center justify-center bg-gray-900 bg-cover bg-no-repeat bg-image">
      <div className="rounded-xl bg-white bg-opacity-50 px-16 py-10 shadow-lg backdrop-blur-md max-sm:px-8">
        <div className="text-white">
          <div className="mb-8 flex flex-col items-center">
            <img src="./favicon.png" width="150" alt="" />
            <h1 className="mb-2 text-2xl">Buy Bikes</h1>
            <span className="text-white">Enter Login Details</span>
            <span className="text-xl text-red-700 font-bold mt-4">{showMessage}</span>
          </div>
          <form onSubmit={loginUser}>
            <div className="mb-4 text-lg">
              <input
                className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
                required
                onChange={(e) => setEmail(e.target.value)}
                value={email}
                type="email"
                name="name"
                placeholder="id@email.com"
              />
            </div>

            <div className="mb-4 text-lg">
              <input
                className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
                required
                onChange={(e) => setPassowrd(e.target.value)}
                value={password}
                type="Password"
                name="name"
                placeholder="*********"
              />
            </div>
            <div className="mt-8 flex justify-center text-lg text-black">
              {loading ? (
                <div className="spinner-border text-primary" role="status">
                  <span className="visually-hidden">
                    {' '}
                    <img src="https://i.stack.imgur.com/kOnzy.gif" className="w-10 h-10" alt="loading" />
                  </span>
                </div>
              ) : (
                <button
                  type="submit"
                  className="rounded-3xl bg-black bg-opacity-50 px-10 py-2 text-white shadow-xl backdrop-blur-md transition-colors duration-300 hover:bg-yellow-600"
                >
                  Login
                </button>
              )}

            </div>
          </form>
          <p className="mt-8 text-xs font-light text-center text-gray-700">
            {' '}
            Do not have an account?
            <a
              href="./signup"
              className="font-medium text-purple-600 hover:underline"
            >
              Sign up
            </a>
          </p>
        </div>
      </div>
    </div>
  );
};

export default LogIn;
