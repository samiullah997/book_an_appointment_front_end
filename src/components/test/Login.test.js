import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/Store';
import LogIn from '../authentication/LogIn';

describe('Login', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><LogIn /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
