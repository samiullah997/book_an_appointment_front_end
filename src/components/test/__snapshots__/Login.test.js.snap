// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-screen w-full items-center justify-center bg-gray-900 bg-cover bg-no-repeat bg-image"
      >
        <div
          class="rounded-xl bg-white bg-opacity-50 px-16 py-10 shadow-lg backdrop-blur-md max-sm:px-8"
        >
          <div
            class="text-white"
          >
            <div
              class="mb-8 flex flex-col items-center"
            >
              <img
                alt=""
                src="./favicon.png"
                width="150"
              />
              <h1
                class="mb-2 text-2xl"
              >
                Buy Bikes
              </h1>
              <span
                class="text-white"
              >
                Enter Login Details
              </span>
              <span
                class="text-xl text-red-700 font-bold mt-4"
              />
            </div>
            <form>
              <div
                class="mb-4 text-lg"
              >
                <input
                  class="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
                  name="name"
                  placeholder="id@email.com"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mb-4 text-lg"
              >
                <input
                  class="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
                  name="name"
                  placeholder="*********"
                  required=""
                  type="Password"
                  value=""
                />
              </div>
              <div
                class="mt-8 flex justify-center text-lg text-black"
              >
                <button
                  class="rounded-3xl bg-black bg-opacity-50 px-10 py-2 text-white shadow-xl backdrop-blur-md transition-colors duration-300 hover:bg-yellow-600"
                  type="submit"
                >
                  Login
                </button>
              </div>
            </form>
            <p
              class="mt-8 text-xs font-light text-center text-gray-700"
            >
               
              Do not have an account?
              <a
                class="font-medium text-purple-600 hover:underline"
                href="/signup"
              >
                Sign up
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-screen w-full items-center justify-center bg-gray-900 bg-cover bg-no-repeat bg-image"
    >
      <div
        class="rounded-xl bg-white bg-opacity-50 px-16 py-10 shadow-lg backdrop-blur-md max-sm:px-8"
      >
        <div
          class="text-white"
        >
          <div
            class="mb-8 flex flex-col items-center"
          >
            <img
              alt=""
              src="./favicon.png"
              width="150"
            />
            <h1
              class="mb-2 text-2xl"
            >
              Buy Bikes
            </h1>
            <span
              class="text-white"
            >
              Enter Login Details
            </span>
            <span
              class="text-xl text-red-700 font-bold mt-4"
            />
          </div>
          <form>
            <div
              class="mb-4 text-lg"
            >
              <input
                class="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
                name="name"
                placeholder="id@email.com"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="mb-4 text-lg"
            >
              <input
                class="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
                name="name"
                placeholder="*********"
                required=""
                type="Password"
                value=""
              />
            </div>
            <div
              class="mt-8 flex justify-center text-lg text-black"
            >
              <button
                class="rounded-3xl bg-black bg-opacity-50 px-10 py-2 text-white shadow-xl backdrop-blur-md transition-colors duration-300 hover:bg-yellow-600"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
          <p
            class="mt-8 text-xs font-light text-center text-gray-700"
          >
             
            Do not have an account?
            <a
              class="font-medium text-purple-600 hover:underline"
              href="/signup"
            >
              Sign up
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
